codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolUnitTestResponseGenerator.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ *
+ */
+
+package software.amazon.smithy.go.codegen.integration;
+
+import java.util.List;
+import java.util.logging.Logger;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.protocoltests.traits.HttpResponseTestCase;
+
+/**
+ * Generates HTTP protocol unit tests for HTTP response test cases.
+ */
+public class HttpProtocolUnitTestResponseGenerator extends HttpProtocolUnitTestGenerator<HttpResponseTestCase> {
+    private static final Logger LOGGER = Logger.getLogger(HttpProtocolUnitTestResponseGenerator.class.getName());
+
+    /**
+     * Builder to lazy initialize the protocol test generator with operation specific configuration.
+     */
+    public static class Builder {
+        /**
+         * Builds the test generator for the provided configuration.
+         *
+         * @param model          Smithy model.
+         * @param symbolProvider Smithy symbol provider.
+         * @param protocolName   Name of the model's protocol that will be generated.
+         * @param operation      shape for the Operation the unit test is generated for.
+         * @param testCases      Set of test cases for the operation.
+         * @return initialize protocol test generator.
+         */
+        public HttpProtocolUnitTestResponseGenerator build(
+                Model model, SymbolProvider symbolProvider, String protocolName, OperationShape operation,
+                List<HttpResponseTestCase> testCases
+        ) {
+            return new HttpProtocolUnitTestResponseGenerator(model, symbolProvider, protocolName, operation, testCases);
+        }
```

Comment:
Usually the builder provide setter functions to set the relevant fields and then the build function takes no arguments and passed the builder to a `HttpProtocolUnitTestResponseGenerator` constructor. If you implement the SmithyBuilder interface you can also assert that certain values were set.

Reply:
Was talking with Kevin about this too. Will update these builders to follow that pattern.